autoconfigure/storage-elasticsearch-aws/src/main/java/zipkin/autoconfigure/storage/elasticsearch/aws/ElasticsearchDomainEndpoint.java

```diff
@@ -13,73 +13,75 @@
  */
 package zipkin.autoconfigure.storage.elasticsearch.aws;
 
-import com.squareup.moshi.JsonReader;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.common.AggregatedHttpRequest;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.HttpStatusClass;
 import java.io.IOException;
 import java.util.Collections;
 import java.util.List;
+import java.util.concurrent.CompletionException;
 import java.util.logging.Logger;
-import okhttp3.HttpUrl;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.Response;
-import okio.Buffer;
-import zipkin2.elasticsearch.ElasticsearchStorage;
+import zipkin2.elasticsearch.ElasticsearchStorage.HostsSupplier;
 
-import static zipkin2.elasticsearch.internal.JsonReaders.enterPath;
+import static zipkin.autoconfigure.storage.elasticsearch.aws.ZipkinElasticsearchAwsStorageAutoConfiguration.JSON;
 
-final class ElasticsearchDomainEndpoint implements ElasticsearchStorage.HostsSupplier {
+final class ElasticsearchDomainEndpoint implements HostsSupplier {
   static final Logger log = Logger.getLogger(ElasticsearchDomainEndpoint.class.getName());
 
-  final OkHttpClient client;
-  final Request describeElasticsearchDomain;
+  final HttpClient client;
+  final AggregatedHttpRequest describeElasticsearchDomain;
 
-  ElasticsearchDomainEndpoint(OkHttpClient client, HttpUrl baseUrl, String domain) {
+  ElasticsearchDomainEndpoint(HttpClient client, String domain) {
     if (client == null) throw new NullPointerException("client == null");
-    if (baseUrl == null) throw new NullPointerException("baseUrl == null");
     if (domain == null) throw new NullPointerException("domain == null");
     this.client = client;
     this.describeElasticsearchDomain =
-        new Request.Builder()
-            .url(baseUrl.newBuilder("2015-01-01/es/domain").addPathSegment(domain).build())
-            .tag("get-es-domain")
-            .build();
+        AggregatedHttpRequest.of(HttpMethod.GET, "/2015-01-01/es/domain/" + domain);
```

Comment:
TODO: name this "get-es-domain" like we did upstream

Reply:
Maybe not worth it - IIUC, this is always going to be client-initiated and not sampled.